pkg/builder/controller.go

```diff
@@ -309,7 +317,14 @@ func (blder *Builder) doController(r reconcile.Reconciler) error {
 
 	// Setup the logger.
 	if ctrlOptions.LogConstructor == nil {
-		log := blder.mgr.GetLogger().WithValues(
+		var log logr.Logger
+		if ctrlOptions.GetDefaultLogger != nil {
+			log = ctrlOptions.GetDefaultLogger()
+		} else {
+			log = blder.mgr.GetLogger()
+		}
+
```

Comment:
IHMO we should have only one log approach and ideally follow the k8s standards.
Why? 

- That helps those that are looking for Oberservaility, many tools will try to check the logs centralized
- Maintainability. We do not increase the burn to have more implementation for logs 
- Encourage standards.

In this way, I'd like to raise a question: What is the k8s format? Is k8s doing the logs using the format of WithLogConstructor or of the removed WithLogger? 

/hold

Reply:
k8s uses [klog](https://github.com/kubernetes/klog), which is derived from good-old glog.
In that case, I think it's better to avoid the use of logr, which backend is configuratble, and we can stick to klog.

Reply:
@timonwong,  

The most important I think is not the dependency used but the format of the logs. 

However, I think would be nice if we could check how the format is outputted by k8s as the dep (which you already checked) and if we can or do not follow the same standards. if not, why not? Also, have we any Kubernetes good practices definition about it? If so, I think we should also follow up on that. 

WDYT? If that also makes sense for you and you would like to work on this I think the next step would be an assessment and open an issue and/or PR with the proposal.

I will 
/hold cancel

to let the others also share their thoughts.

Reply:
Hi @timonwong, 

(ihmo) The most important I think is not the dependency used but the output/format of the logs. According to the description of https://github.com/kubernetes-sigs/controller-runtime/pull/1827 the goal is to align with `align to Kubernetes structured logging, add reconcileID ` 

However, I think would be nice if we could check how the format is outputted by k8s as the dep (which you already checked) and if we can or do not follow the same standards. if not, why not? Also, have we any Kubernetes good practices definition about it? If so, I think we should also follow up on that. By last, I think also would be nice to understand why we are using `GetDefaultLogger ` and not klog. What was the motivation for this adoption instead of klog?  

(IMHO) the above assessment would need to be done before we propose changes on how it is implemented. If we figure out that we need to change then, maybe open an issue describing the proposal and motivations for that can help out in the discussion.  Also, a PR proposing the changes based on the motivations can be helpful for sure too. 

WDYT?

Reply:
@camilamacedo86 I think a lot of what you are asking here is orthogonal to this change. This changes goal is to have the same fields that get set when a logger comes from the mgr when a custom logger is used.

I agree that it is not great to have to ways to set a custom logger that are different in a very subtle way.